src/psyclone/tests/psyGen_test.py

```diff
@@ -2264,3 +2264,18 @@ def test_find_w_args_multiple_deps():
     # each of the indices are unique (otherwise the set would be
     # smaller)
     assert len(indices) == vector_size
+
+
+def test_kern_ast():
+    ''' Test that we can obtain the fparser2 AST of a kernel. '''
+    from psyclone.gocean1p0 import GOKern
+    from fparser.two import Fortran2003
+    _, invoke = get_invoke(os.path.join("gocean1p0",
+                                        "nemolite2d_alg_mod.f90"),
+                           api="gocean1.0", idx=0)
+    sched = invoke.schedule
+    kern = sched.children[0].children[0].children[0]
+    assert isinstance(kern, GOKern)
+    assert kern.ast
+    assert isinstance(kern.ast, Fortran2003.Program)
+
```

Comment:
Perhaps add a check that the subroutine name is the one we were expecting?

Reply:
Actually, `kern.ast` holds the AST for the whole Fortran module so I don't think that that check is relevant.

Reply:
OK.